### ISSUE TITLE
"No response error" UnhandledPromiseRejection with createWsServer crashes nodejs server.

### ISSUE THREAD

**Issue by aurmartin** [tinyplex/tinybase#194]
### Describe the bug

I am having UnhandledPromiseRejection errors on the server when using [createWsServer](https://tinybase.org/api/synchronizer-ws-server/functions/creation/createwsserver/) with persistence server-side. The error crashes the whole node process:

```
UnhandledPromiseRejection: This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason "No response from TQjiy13/0Fkji6fv3Zk3tA== to G0NvmWywJSsz7C-e, 4".
    at throwUnhandledRejectionsMode (node:internal/process/promises:389:7)
    at processPromiseRejections (node:internal/process/promises:470:17)
    at process.processTicksAndRejections (node:internal/process/task_queues:96:32) {
  code: 'ERR_UNHANDLED_REJECTION'
}
```

I can reproduce it using the same code as the Todo collaboration demo, see the linked repository.

This error occurs frequently, at random moments.

I can reproduce it systematically when I open the app in one browser, then in another browser and then close the second browser.

The error also appear client-side:

![image](https://github.com/user-attachments/assets/72c50521-4120-400a-8535-0add1d53320b)

### Your Example Website or App

https://github.com/aurmartin/tinybase-bug-report

### Steps to Reproduce the Bug or Issue

1. Clone the repo https://github.com/aurmartin/tinybase-bug-report
2. Start the react app and the server (see the README.md)
3. Open the react app, populate the store.
4. Open the react app in another browser, quickly refresh after the start.

### Expected behavior

The server and the application should keep working when a client close the connection.

### Platform

- OS: macOS
- Browser: Brave v1.71.121
- Tinybase: v5.3.7
- Node.js: v20.18.0

**Comment by jamesgpearce** [comment]
Whoah - good one! I think your clients are in the middle of a conversation when one is closed, so the other times out. This should be easy to repro and handle correctly. Let me take a look.

**Comment by jamesgpearce** [comment]
OK I think there are two issues here. The first is that because this app uses StrictMode, the main component renders twice, and so two sockets get created, both of which the server sees. The first disappears (perhaps too quickly) and a race condition ensues.

The second issue is that that error is not getting handled. I can fix that!

I will try to address both, but for immediate gratification, just remove the StrictMode wrapper in main.jsx!

**Comment by aurmartin** [comment]
Thanks a lot for your responsiveness on this issue!

Yes, I have noticed the weird behavior with the React StrictMode.

But event without it, I have the issue with two clients in two different browsers. It is harder to reproduce but still happen frequently.

For the moment, I switched to a Yjs persister and the Yjs websocket provider. It seems more robust, like reconnecting the websocket automatically when the server restarts.

Moreover, the fact that the whole mergeable stored must be stored in a single JSON file, won't scale well in my case, I think.
